### PR TITLE
fix: inject SKILL.md content, load trigger fixtures, pass MCP servers

### LINKED ISSUE #180: mcp_servers config section seems do not start/connect MCP server — agent has zero tool calls
### Summary

When configuring an MCP server via the `mcp_servers` section in `eval.yaml`, the MCP server is not connected to the agent session. The agent **can** make regular tool calls (`view`, `edit`, `powershell`, etc.) but **cannot** use any MCP tools (e.g. `azsdk_run_typespec_validation`, `azsdk_typespec_generate_authoring_plan`).

The agent falls back to running `tsp compile .` via `powershell` instead of using the MCP tool `azsdk_run_typespec_validation`, which causes the `mcp_workflow` grader to fail.

### Related Issue

- #165 (diff grader issues — same eval suite, different problem)

### Environment

- waza version: latest
- OS: Windows
- Model: claude-opus-4.6-1m

### Eval Config

I'm referring to this document to set the `mcp_servers` section. https://github.com/microsoft/waza/blob/main/docs/INTEGRATION-TESTING.md
```yaml
config:
  trials_per_task: 1
  timeout_seconds: 1000
  parallel: false
  executor: copilot-sdk
  model: claude-opus-4.6-1m
  mcp_servers:
    azure-sdk-mcp:
      type: stdio
      command: pwsh
      args: ["C:\\workspace\\azure-rest-api-specs\\eng\\common\\mcp\\azure-sdk-mcp.ps1", "-Run"]
```

### Observed Behavior

The eval report shows the agent made **19 tool calls** but **none were MCP tools**:

```
tools_used: [skill, view, view, view, view, view, view, view, view, view, view, web_fetch, view, edit, powershell, powershell, powershell, powershell, view]
```

Key observations:
- The agent correctly invoked the `azure-typespec-author` skill (`routing_check` passed)
- The agent used `powershell` to run `tsp compile .` manually instead of using the MCP tool `azsdk_run_typespec_validation`
- **No MCP tools appear in the tools_used list**, confirming the MCP server was not connected

### Grader Results

| Grader | Type | Score | Passed | Feedback |
|--------|------|-------|--------|----------|
| `routing_check` | skill_invocation | 1.0 | ✅ | Skill invocation sequence matched |
| `mcp_workflow` | action_sequence | 0.095 | ❌ | Expected `[edit, azsdk_run_typespec_validation]` in order, but `azsdk_run_typespec_validation` never appeared |
| `version_enum_check` | diff | 0.33 | ❌ | Missing expected fragments (related to #165) |

The `mcp_workflow` grader details:
```json
{
  "actual_actions": ["skill", "view", "view", "view", "view", "view", "view", "view", "view", "view", "view", "web_fetch", "view", "edit", "powershell", "powershell", "powershell", "powershell", "view"],
  "expected_actions": ["edit", "azsdk_run_typespec_validation"],
  "matching_mode": "in_order_match",
  "precision": 0.053,
  "recall": 0.5
}
```

The `edit` action was found (recall 0.5) but `azsdk_run_typespec_validation` was never called because the MCP server wasn't available.

### Expected Behavior

The `mcp_servers` section should:
1. Start the MCP server process (via `pwsh ... azure-sdk-mcp.ps1 -Run`)
2. Connect it to the agent session so MCP tools are available as callable tools
3. The agent should be able to invoke MCP tools like `azsdk_run_typespec_validation`

### Summary Stats

```json
{
  "total_tests": 1,
  "succeeded": 0,
  "failed": 1,
  "success_rate": 0,
  "aggregate_score": 0.476,
  "duration_ms": 171610,
  "usage": {
    "turns": 13,
    "input_tokens": 307342,
    "output_tokens": 3385,
    "cache_read_tokens": 220527,
    "premium_requests": 78
  }
}
```

### Questions

1. Is `mcp_servers` the correct way to provide MCP servers to the agent in evals?
2. Does the `type: stdio` + `command: pwsh` + `args` format work correctly on Windows?
3. Is there any logging I can enable to see whether the MCP server process is actually spawned and connected?
4. Are MCP tools supposed to appear in the `tools_used` list in the session digest?

### Full report
[claude-opus-4.6-1m.json](https://github.com/user-attachments/files/26397024/claude-opus-4.6-1m.json)

### LINKED ISSUE #186: Trigger tests don't seem to have access to the fixtures directory
I ran the example code-exaplainer trigger tests and read the output. The transcript of all the trigger tests indicate the workspace gave to these trigger tests are empty. This causes all the trigger tests to not invoke this example skill. On the other hand, task test runs seem to have access to a copy of the files in the fixtures directory since the agent output includes meaningful information of the relevant files.

### LINKED ISSUE #190: copilot-sdk executor does not inject SKILL.md as system context
## Problem

When running evals with `executor: copilot-sdk`, the model responds as a generic assistant — it has no awareness of the skill's SKILL.md content. The skill files (SKILL.md + references/) are not injected into the model's system prompt or context.

## Evidence

Running `waza run aspire-orchestration --model gpt-4.1 --verbose` against an Aspire skill that teaches the model to use `aspire start` instead of `dotnet run`:

```
[PROMPT] Start my Aspire app
[RESPONSE] I couldn't find an obvious Aspire app entrypoint... 
Please let me know the name of your main project file...
```

```
[PROMPT] How do I restart just the API service?
[RESPONSE] To restart just the API service, you typically use systemctl...
```

The model should be responding with `aspire start` and `aspire resource <name> restart` — the exact guidance in the SKILL.md. Instead, it gives generic answers because it never sees the skill content.

## Expected Behavior

The executor should inject the skill's SKILL.md (and optionally referenced files from `references/`) into the model's system prompt or as context, so the model's responses are guided by the skill's rules and decision tables.

The verbose output shows `Skill Directories` are detected:
```
Skill Directories:
  - /path/to/skills/aspire-orchestration
```

But the content from those directories is not making it into the LLM call.

## Reproduction

```bash
# 1. Any skill repo with SKILL.md
waza run <skill-name> --model gpt-4.1 --verbose --task <any-task>

# 2. Observe: responses are generic, not skill-aware
# 3. The SKILL.md content is never shown in the prompt
```

## Environment

- waza v0.23.0
- macOS (darwin/arm64)
- executor: copilot-sdk
- model: gpt-4.1

## Workaround

None currently. The `mock` executor doesn't have this issue because it doesn't call an LLM, but it can't validate real model behavior.

## Impact

This blocks eval-driven skill development — the core waza workflow. Without skill injection, evals only measure the base model's knowledge, not the skill's effectiveness at improving the model's responses.

### PR DESCRIPTION
**Pull request by spboyer**
## Summary

Fixes three related bugs that prevent waza evals from working correctly with the copilot-sdk executor.

### #190 — SKILL.md not injected as system context
**Root cause:** PR #140 removed `buildSkillSystemMessage()`, relying solely on `SkillDirectories`. The SDK's `SkillDirectories` handles skill *tool routing* but doesn't inject skill content into the system prompt.

**Fix:** Re-added skill content injection via `SessionConfig.SystemMessage` (mode: `append`). Injects the full SKILL.md body for the target skill and compact `<available_skills>` summaries for all discovered skills. Scans both direct and nested skill directories, skipping hidden/vendor dirs.

### #186 — Trigger tests don't have access to fixtures
**Root cause:** `trigger/runner.go:testTrigger()` built `ExecutionRequest` without `Resources`. The workspace was always empty.

**Fix:** `NewRunner()` now walks `cfg.FixtureDir()` once at init, caches `[]ResourceFile`, and passes them on every trigger `ExecutionRequest`. Skips hidden dirs, vendor, node_modules, and files >1MB.

### #180 — mcp_servers config not passed to copilot session
**Root cause:** `spec.Config.ServerConfigs` was never passed through to `ExecutionRequest` or the copilot `SessionConfig`.

**Fix:** Added `MCPServers` field to `ExecutionRequest`, `convertMCPServers()` helper (validates each entry is map-shaped), and passthrough in both orchestration and trigger runners.

## Test Coverage
- 8 new unit tests for `buildSkillSystemMessage` and `parseSkillFrontmatter`
- 3 new copilot engine tests verifying `SystemMessage` and `MCPServers` on CreateSession and ResumeSession
- 6 new trigger runner tests for fixture loading, MCP propagation, and edge cases
- All existing tests pass (`go test ./...`, `go vet`)

Closes #190
Closes #186
Closes #180

### CODE REVIEW

File: internal/trigger/runner.go [hunk 1]
```diff
@@ -165,6 +174,88 @@ func (r *Runner) testTrigger(ctx context.Context, prompt string) (*execution.Exe
...
+		SourceDir:  r.cfg.SpecDir(),
+		Resources:  r.fixtures,
+		MCPServers: r.mcpConfig,
 		Timeout:    time.Duration(timeout) * time.Second,
 	})
 }
+
+// loadFixtureDir recursively walks a fixture directory and returns all files
+// as ResourceFiles. Skips hidden dirs, node_modules, vendor, and binary files.
+// Returns nil if dir is empty or doesn't exist.
```
[the remark this answers is not in the document]
Comment (spboyer) [reply internal/trigger/runner.go:186]: Good catch — the doc comment overstates the implementation. The >1MB size filter is the actual guard. Will fix the comment as a follow-up.

File: internal/trigger/runner.go [hunk 2]
```diff
@@ -165,6 +174,88 @@ func (r *Runner) testTrigger(ctx context.Context, prompt string) (*execution.Exe
...
+	if len(serverConfigs) == 0 {
+		return nil
+	}
+
+	result := make(map[string]copilot.MCPServerConfig, len(serverConfigs))
+	for name, cfg := range serverConfigs {
+		cfgMap, ok := cfg.(map[string]any)
+		if !ok {
+			fmt.Fprintf(os.Stderr, "Warning: mcp_server %q config is not a map, skipping\n", name)
+			continue
```
[the remark this answers is not in the document]
Comment (spboyer) [reply internal/trigger/runner.go:256]: Acknowledged — the stderr warning matches the pattern used elsewhere in the orchestration runner (e.g., loadResources). Will consolidate logging in a follow-up.

File: internal/orchestration/runner.go [hunk 3]
```diff
@@ -1205,6 +1208,25 @@ func (r *TestRunner) loadResources(tc *models.TestCase) []execution.ResourceFile
...
+	for name, cfg := range serverConfigs {
+		cfgMap, ok := cfg.(map[string]any)
+		if !ok {
+			fmt.Fprintf(os.Stderr, "Warning: mcp_server %q config is not a map, skipping\n", name)
+			continue
+		}
+		result[name] = copilot.MCPServerConfig(cfgMap)
+	}
+	return result
+}
```
[the remark this answers is not in the document]
Comment (spboyer) [reply internal/orchestration/runner.go:1228]: Valid point on duplication. These are intentionally separate since trigger and orchestration packages are independent. Will consider extracting to a shared utils package in a follow-up.

File: .github/workflows/squad-promote.yml [hunk 4]
```diff
@@ -49,7 +49,6 @@ jobs:
             .ai-team/ \
             .squad/ \
             .ai-team-templates/ \
-            .squad-templates/ \
             team-docs/ \
             "docs/proposals/" || true
```
[the remark this answers is not in the document]
Comment (spboyer) [reply .github/workflows/squad-promote.yml:53]: This file is not part of the changes in this PR — it appears in the diff due to branch divergence. No changes were made to squad-promote.yml.

File: .adc-sdk/adc.go [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+//	// Create a sandbox
+//	sandbox, err := client.Sandboxes.CreateFromDiskImage(ctx, models.CreateFromDiskImageOptions{
+//		DiskImage: models.SandboxSourceDiskImage{ID: diskImage.ID},
+//		CPU:       "1000m",
+//		Memory:    "1024Mi",
+//	})
+//
+//	// Execute a command
+//	result, err := sandbox.ExecuteShellCommand(ctx, "echo hello", "", nil, "")
+package adc
```
[the remark this answers is not in the document]
Comment (spboyer) [reply .adc-sdk/adc.go:30]: The .adc-sdk/ and .squad/ files are not part of this PR's changes — they appear in the diff due to branch history. This PR only touches internal/execution, internal/orchestration, and internal/trigger.

File: internal/execution/copilot.go [hunk 6]
```diff
@@ -407,3 +421,134 @@ func allowAllTools(request copilot.PermissionRequest, invocation copilot.Permiss
...
+	}
+
+	// Summary block for all discovered skills
+	sb.WriteString("\n<available_skills>\n")
+	for _, s := range skills {
+		sb.WriteString("<skill>\n")
+		fmt.Fprintf(&sb, "  <name>%s</name>\n", s.Name)
+		if s.Description != "" {
+			fmt.Fprintf(&sb, "  <description>%s</description>\n", s.Description)
+		}
```
[the remark this answers is not in the document]
Comment (spboyer) [reply internal/execution/copilot.go:491]: Low risk in practice — skill names come from YAML frontmatter and are typically kebab-case identifiers. The original code (pre-PR#140) had the same pattern. Will add escaping in a follow-up for defense in depth.
